src/Hamiltonians/geometry.jl

```diff
@@ -1,256 +1,227 @@
 """
-    abstract type LatticeGeometry{D}
+    Geometry(dims::NTuple{D,Int}, fold::NTuple{D,Bool})
 
-A `LatticeGeometry` controls which sites in an [`AbstractFockAddress`](@ref) are considered
-to be neighbours.
+Represents a `D`-dimensional grid. Used to convert between cartesian vector indices (tuples
+or `SVector`s) and linear indices (integers). When indexed with vectors, it folds them back
+into the grid if the out-of-bounds dimension is periodic and 0 otherwise (see example
+below).
```

Comment:
```suggestion
Represents a `D`-dimensional grid. Used to define a cubic lattice and boundary conditions 
for some [`AbstractHamiltonian`](@ref)s. The type instance can be used to convert between 
cartesian vector indices (tuples or `SVector`s) and linear indices (integers). When indexed 
with vectors, it folds them back into the grid if the out-of-bounds dimension is periodic and 
0 otherwise (see example below).
```